src/storage.rs

```diff
@@ -1038,7 +1041,10 @@ pub async fn retain_periodically(
                 }
                 info!("Database cleanup completed.");
                 {
-                    let mut running_flag = running_flag.lock().unwrap();
+                    let mut running_flag = running_flag.lock().unwrap_or_else(|e| {
+                        warn!("Poisoned Mutex. Recovering the boolean flag.");
```

Comment:
I think we can use AtomicBool type instead of `running_flag`. I think we can avoid potential panicking from the mutex behavior, and achieve simpler code.